### ISSUE TITLE
Add the ability to message on unexpected success of "negative verification" problems

### ISSUE THREAD

**Issue by wrwg** [boogie-org/boogie#348]
We have some use cases where we want to create "negative" verification problems, that is, situations where we consider failure of verification success, and passing verification a failure.

Consider the following example:

```
procedure {:inline 1} some_stuff() {
  <<some complex Boogie with assumes and asserts, handwritten or generated>>
}

procedure some_stuff_$verify() { call some_stuff(); }

procedure some_stuff_$soundness { call some_stuff(); assert false; }
```

Here `some_stuff_$soundness` is a sanity check whether the specifications for `some_stuff` did not introduce any unsound assumptions. We expect that an `assert false` fails it this not the case, but otherwise may pass verification.

Currently, if a function passes a verification, we never get any direct feedback from this from Boogie. It would be nice to have an attribute `{:print_if_verifies}` which can be used as such in this example:

```
procedure {:print_if_verifies "ERROR: soundness check failed"} some_stuff_$soundness { call some_stuff(); assert false; }
```

@shazqadeer @junkil-park

**Comment by shazqadeer** [comment]
Done.

https://github.com/boogie-org/boogie/blob/master/Test/test13/msg.bpl

**Comment by RustanLeino** [comment]
Hi @wrwg, Awesome! What you're describing sounds similar to Boogie's `-smoke` option. By adding the `assert false;` commands yourself, I imagine you're getting something more tailored for your purposes. If your scheme is something that Boogie could do for you and could be generally useful, then you're welcome to contribute it to Boogie.

**Comment by wrwg** [comment]
@RustanLeino Wow, I was not aware of the `-smoke` option. I also see it comes with a different timeout which is precisely also we figured is needed after we implemented this ourselves.

In general, we want to do more negative verification than just assert false, so this feature is still useful. For example, we are thinking of doing some kind of "mutation testing" on specs and programs to test specification coverage. 

The ugly thing though is that those negative verification problems tend to time out easily if they find the expected behavior (verification failure), so we are not sure yet whether this will be a viable path. For now, we are moving this kind of checks from a PR blocker to nightly because of the timeout behavior.
